src/formatDate.js

```diff
@@ -48,9 +48,67 @@
  *
  * @returns {string}
  */
+const DAY_SIGN = 'DD';
+const MONTH_SIGN = 'MM';
+const YEAR_SIGN = 'YYYY';
+const YEAR_SIGN_SHORT = 'YY';
+const CENTURY_19 = '19';
+const CENTURY_20 = '20';
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldFormatDate = date.split(fromFormat[3]);
+  const newFormatDate = [];
+  let newYearLength = 0;
+  let oldYearLength = 0;
+  let year,
+    month,
+    day,
+    yearIndex;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i].includes(YEAR_SIGN_SHORT)) {
+      year = oldFormatDate[i];
+      oldYearLength = fromFormat[i].length;
+    }
+
+    if (fromFormat[i].includes(MONTH_SIGN)) {
+      month = oldFormatDate[i];
+    }
+
+    if (fromFormat[i].includes(DAY_SIGN)) {
+      day = oldFormatDate[i];
+    }
+  }
```

Comment:
We refer to `fromFormat[i]` and `oldFormatDate[i]` many times, so this could be put into a separate variables.